### DISCUSSION TITLE
Google API required by the meeting detail page to show map.

### DISCUSSION THREAD

**Discussion by ghost** [code4recovery/12-step-meeting-list#1452] [Q&A] [answered]
What are the explicitly required google api's required (over and above the javascript api) for the meeting detail page to show a map.

I guess my real question is; does google maps still work with tsml or has everybody just migrated to mapbox?

**Comment by joshreisner** [comment]
just the `Maps JavaScript API`

it still works! i just created a key and applied it at https://code4recovery.org/meetings?tsml-view=map

**Reply by ghost** [reply]
When I use a mapbox key this works. If I supply a valid google api key it does not work, the option for list/map does not show up and the map is not displayed in the meeting detail page.

I have used ACF to display google maps in my front page of my site using the google api key which works. 

Note: If I inspect the meeting detail page its not loading the google maps js. 

If it helps if I use the suffix provided in link above the list is displayed and not the map (but selecting map does work)

And I have removed the restriction on the hosts names that can call the google api.

**Reply by joshreisner** [reply, marked as answer]
are you using TSML UI? that is Mapbox-only. only the legacy interface supports Google Maps

**Reply by ghost** [reply]
Yes I am. We have a number of online meetings etc which display better. Any reason for not using google?

**Reply by joshreisner** [reply]
yes, Mapbox has better pricing, and we often hear concerns from members about Google's privacy policy

it helps our developer bandwith and overall JS bundle size to support a single solution

**Reply by ghost** [reply]
If you are in a larger area the pricing would be an understandable concern. 

Before joining the fellowship I was in software, so you know this reply has been rewritten a number of times ;)

I have spent quite some time trying to work out what was wrong with the google key.

I'm thinking maybe adding validation around the selection of 'User Interface Display' e.g. Hiding the Google API textbox when not valid would be great help for others.

**Reply by joshreisner** [reply]
ah, sorry about that. we do have a message that pops up if you have a google key and change your appearance to TSML UI , but when you come back to the page we don't have that.

<img width="1552" alt="Screenshot 2024-06-20 at 1 42 13 PM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/bd7350c5-580b-4b21-bb6c-4685ad38dfbe">

i'll create a ticket to move the message to be more persistent and put it down in the map keys box
